### ISSUE TITLE
[Experiment Feature]: TypeScript config files support - Node.js v22

### ISSUE THREAD

**Issue by Mister-Hope** [Zhengqbbb/cz-git#192]
### ✅ Checklist

- [X] I am using the **latest version**
- [X] I have searched [FAQ](https://cz-git.qbb.sh/faq/)

### 📄 System Info

```Shell
not needed
```

### 🔎 Description

See https://commitlint.js.org/reference/configuration.html

So https://cz-git.qbb.sh/config/#typescript-template should be fixed

**Comment by Zhengqbbb** [comment]
[commitlint](https://github.com/conventional-changelog/commitlint/blob/be0197fa50ee4cbe6969f0ce30954f87f2455a82/%40commitlint/load/package.json#L54) Replace ts-node with cosmiconfig-typescript-loader package https://github.com/conventional-changelog/commitlint/pull/3722

Although the speed has improved, adding cz-git and czg will double the size, and I am still considering it. 😶‍🌫️

<img width="992" alt="image" src="https://github.com/user-attachments/assets/627262f2-36fd-4f80-b021-be28a7ea1c7a">

**Comment by Mister-Hope** [comment]
1. `commitlint.config.ts` is supported by commitlint directly, so the docs need to be corrected, there is nothing to do with your pacakge
2. Even if you want to have `cz.config.ts` supported, you should externalize `cosmiconfig-typescript-loader`, tools like rollup and tsup can do that easily

**Comment by Mister-Hope** [comment]
Besides, I think you have misunderstood my original issue complaining about the dep size.

Since your pacakge is strongly relying on commitlint, so you should try to reuse package that is required by it as much as possible while keeping the same major, this way you can externalize them safely as they are supposed to be installed to workspace anyway.

The main target for tool package like this is trying to avoid requiring any new package that is heavy. A package is heavy means that it has too many deps or a lot of unused features for your usage. You can try to bundle packages with cz-git, only if they are likely not required by any other packages, or in another word "unpopular"

**Comment by Zhengqbbb** [comment]
Thank you for your reply.
1. However, I don't understand with my documentation not supporting `commitlint.config.ts` Please feel free to point it out.
> Since your package is strongly relying on commitlint, so you should try to reuse package
2. Like `@commitlint/load` package have too many dep [package.json](https://github.com/conventional-changelog/commitlint/blob/master/%40commitlint/load/package.json). I have try but cjs project treeshakable effect is not well
3. Although I do not agree that a lightweight CLI tool, which requires the fastest startup speed, should resolve the ts configuration file at this time, I will keep an open opinion and will not add it in the short term.

**Comment by Mister-Hope** [comment]
> However, I don't understand with my documentation not supporting commitlint.config.ts Please feel free to point it out.

![image](https://github.com/user-attachments/assets/49bc0b39-18d5-4c52-8e08-60ecc3652a35)
![image](https://github.com/user-attachments/assets/4c766801-c861-4ea7-9cf3-bebc1b1de4d3)

No offense, but as I am pointing out the link in issue body, it should not be hard to find these.

**Comment by Zhengqbbb** [comment]
Done
Btw, I will try to configure loading for TypeScript using node > 22 in the future, instead of adding extra dependencies.

**Comment by Zhengqbbb** [comment]
Hi there.
Node.js LTS v22 has been released, and in v1.11.0, I will add the loading of ESM configuration and TS configuration. 
Regarding the loading of TS configuration, it is done using the native Node loading method, but it is currently an experimental feature that requires additional environment injection. 
It is recommended to first use the ESM JS method for transition.

docs preview: https://deploy-preview-197--cz-git.netlify.app/
---

I have released a next pre-release version 🌟(v1.11.0-beta.1)🌟 for testing.
I expect to officially release v1.11.0 next weekend.
Welcome to try it out! 👀 Let me know if there are still any issues. 🎉

---

I am testing in the repository https://github.com/vuepress-theme-hope/vuepress-theme-hope

### Suggestion (ESM js)
1. rename `commitlint.config.ts` => `commitlint.config.js`

```js
import { execSync } from "node:child_process";
import { readdirSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

import { defineConfig } from "cz-git";

const packages = readdirSync(
  join(dirname(fileURLToPath(import.meta.url)), "./packages/"),
);

const scopeComplete = execSync("git status --porcelain || true")
  .toString()
  .trim()
  .split("\n")
  .find((r) => ~r.indexOf("M  packages"))
  ?.replace(/\//g, "%%")
  ?.match(/packages%%((\w|-)*)/)?.[1];

export default defineConfig({
  extends: ["@commitlint/config-conventional"],
  rules: {
    "scope-enum": [2, "always", ["demo", "release", ...packages]],
  },
  prompt: {
    defaultScope: scopeComplete,
    customScopesAlign: !scopeComplete ? "top" : "bottom",
    allowCustomIssuePrefix: false,
    allowEmptyIssuePrefix: false,
  },
});
```

![CleanShot 2024-11-01 at 21 51 25@2x](https://github.com/user-attachments/assets/689e1b6e-25d4-463d-988f-f6db731b1075)

### (TypeScript configure)

```diff
-     "commit": "pnpm git-cz",
+     "commit": "cross-env NODE_OPTIONS=\"--experimental-transform-types --disable-warning ExperimentalWarning\" pnpm git-cz",
```

![CleanShot 2024-11-01 at 21 53 34@2x](https://github.com/user-attachments/assets/f45df1d9-de11-4549-9775-6efa0e54b838)

**Comment by Zhengqbbb** [comment]
The latest version 🌟(v1.11.0)🌟 released and can use Node v22 to loading ts configure, But it is recommended to use ESM js configure as a transition for the future TypeScript configuration file, and switch to it after Node.js runs TypeScript stably.

Docs Link: https://cz-git.qbb.sh/config/#typescript-template
